ziprofs.py

```diff
@@ -198,37 +183,46 @@ def readdir(self, path, fh):
         result.extend(subdirs)
         return result
 
-
     def release(self, path, fh):
-        zip_path = self.get_zip_path(path)
-        if not zip_path:
-            return os.close(fh)
-        return 0
+        if self.get_zip_path(path):
+            f = self._zip_file_fh[fh]
+            with self._zip_lock_fh[fh]:
+                ret = f.close()
+                del self._zip_file_fh[fh]
+                del self._zip_lock_fh[fh]
```

Comment:
Guard this by a lock as above to make sure both entries are deleted atomically.

Reply:
This is outdated, I deleted this line.